programs/stake/src/stake_state.rs

```diff
@@ -166,6 +166,24 @@ impl Meta {
         }
         Ok(())
     }
+
+    pub fn reduce_overage_rent_exempt_reserve(
+        &mut self,
+        rent: &Rent,
+        data_len: usize,
+    ) -> Option<(u64, u64)> {
+        let corrected_rent_exempt_reserve = rent.minimum_balance(data_len);
+        if corrected_rent_exempt_reserve != self.rent_exempt_reserve {
+            // we forcibly update rent_excempt_reserve even if rent_exempt_reserve > account
+            // balance, hoping user might restore rent_exempt status by depositing.
+            // TODO: split is overflow safe?
```

Comment:
What is your concern here?

Reply:
Given large stake account split from small stake account, this rewrite logic could create a situation where `newly rewritten rent exempt reserve` > `stake account balance`.

In that case, further splitting the rewritten account could trigger integer overflow here, maybe?:

https://github.com/solana-labs/solana/blob/3827739708962b0e46cde6b604eea0a03292664c/programs/stake/src/stake_state.rs#L913

Reply:
Fixing here: https://github.com/solana-labs/solana/pull/13649